tests/functests/instancetype_test.go

```diff
@@ -437,3 +472,54 @@ func expectSSHToRunCommand(virtClient kubecli.KubevirtClient, vmName, username s
 		g.Expect(err).ToNot(HaveOccurred())
 	}, defaultVMReadyTimeout, 10*time.Second).Should(Succeed())
 }
+
+func GetDeprecatedFieldFromCrd(virtClient kubecli.KubevirtClient, crdVersion string) []string {
+	crd, err := virtClient.ExtensionsClient().ApiextensionsV1().CustomResourceDefinitions().Get(
+		context.Background(), crdVersion, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	var nonDeprecatedVersion apiextensionsv1.CustomResourceDefinitionVersion
+	for _, version := range crd.Spec.Versions {
+		if !version.Deprecated {
+			nonDeprecatedVersion = version
+		}
+	}
+	schema := nonDeprecatedVersion.Schema.OpenAPIV3Schema
+
+	var deprecatedFields []string
+	findDeprecatedFields(schema, "", &deprecatedFields)
+	return deprecatedFields
+}
+
+func findDeprecatedFields(schema *apiextensionsv1.JSONSchemaProps, path string, result *[]string) {
+	for propName, propSchema := range schema.Properties {
+		fullPath := propName
+		if path != "" {
+			fullPath = path + "." + propName
+		}
+		if strings.Contains(strings.ToLower(propSchema.Description), "deprecated") {
+			*result = append(*result, fullPath)
+		}
+
+		findDeprecatedFields(&propSchema, fullPath, result)
+	}
+}
+
+func isFieldUsed(obj interface{}, path string) bool {
+	val := reflect.ValueOf(obj)
+	for _, part := range strings.Split(path, ".") {
+		if val.Kind() == reflect.Ptr {
+			val = val.Elem()
+		}
+		if val.Kind() != reflect.Struct {
```

Comment:
Is the last part of your path not always a field, e.g. a string or something?

Reply:
it might be `string`/`slice`/`nil`

Reply:
That means a field not being a struct can never be "used"?